src/v2/ion/responseTransformer.js

```diff
@@ -170,31 +170,31 @@ const convertRedirectIdPToSuccessRedirectIffOneIdp = (settings, result, lastResu
   }
 };
 
-
 /**
- * API reuses `redirect-idp` remeditaion form for IdP Authenticator for both verify and enroll.
- * Hence IdP Authenticator becomes outlier comparing with other Authenticators in terms of
+ * API reuses `redirect-idp` remediation form for PIV IdP and IdP Authenticator.
+ * IdP Authenticator becomes outlier comparing with other Authenticators in terms of
  * using `challenge-authenticator` and `enroll-authenticator` remediation form.
+ * The UX for PIV IdP is different from other idps in terms of the PIV
+ * instructions view that needs to be rendered before we redirect to mtls.
  *
- * This function change `redirect-idp` to `challenge-authenticator` or `enroll-authenticator`
- * for IdP Authenticator.
+ * This function changes `redirect-idp` to `challenge-authenticator` or `enroll-authenticator`
+ * for IdP Authenticator and changes `redirect-idp` to `piv-idp` for PIV IdP.
  */
-const modifyFormNameForIdPAuthenticator = result => {
+const modifyFormNameForIdP = result => {
   if (Array.isArray(result.remediations)) {
-    const idpAuthenticator = result.remediations.filter(
-      remediation => {
-        return remediation.name === RemediationForms.REDIRECT_IDP
-          && remediation?.relatesTo?.value?.key === 'external_idp';
+    result.remediations.forEach(remediation => {
+      if (remediation.name === RemediationForms.REDIRECT_IDP &&
+          remediation?.relatesTo?.value?.key === 'external_idp') {
+        // idp authenticator
+        const isVerifyFlow = Object.prototype.hasOwnProperty.call(result, 'currentAuthenticatorEnrollment');
+        idpAuthenticator[0].name = isVerifyFlow ? 'challenge-authenticator' : 'enroll-authenticator';
       }
-    );
-
-    if (idpAuthenticator.length === 1) {
-      const isVerifyFlow = Object.prototype.hasOwnProperty.call(result, 'currentAuthenticatorEnrollment');
-      idpAuthenticator[0].name = isVerifyFlow ? 'challenge-authenticator' : 'enroll-authenticator';
-    }
-
+      if (remediation.name === RemediationForms.REDIRECT_IDP && remediation.type === IDP_FORM_TYPE.X509) {
```

Comment:
nit: Can we restructure this a little bit to only check for `RemediationForms.REDIRECT_IDP` once (ex: `return` when `remediation.name !== RemediationForms.REDIRECT_IDP`)? I'm assuming the more flows we add, we'll need to alter the name of the remediation more often:

Reply:
+1
Also, before we were using `filter` and then checking if the result had `length==1`, otherwise we didn't change anything. With these changes, we will change the remediation form name for all `external_idp` remediations (in case there are more than 1, is that possible?). This sounds good to me, but wanted to bring it up in case this is something we don't need to do.

Reply:
I just changed this logic a bit based on Pradeep's comments. Can you please take another look @mauriciocastillosilva-okta @jmelberg-okta ?

Reply:
looks good, just the same thing I brought up earlier: we are now changing the name of the form for ALL forms, and before we were doing this only if there was only 1 `external_idp`. I still think this is good, but just want to confirm

Reply:
@jmelberg-okta is this ok? I do not think there can be multiple idp authenticators right now, right?

Reply:
there will be only one remediation form..
https://github.com/okta/okta-signin-widget/blob/master/src/v2/models/AppState.js#L252